src/views/ProtectionHome.vue

```diff
@@ -146,16 +146,29 @@ export default class ProtectionHome extends BaseComponent {
         x => x.fullCoverage - now > 0
       );
       return !isFullyProtected;
-    } else {
-      const groupedPos = groupPositionsArray(vxm.ethBancor.protectedPositions);
-      const fullRow = groupedPos.find(x => x.id === row.id);
-      if (fullRow) row.collapsedData = fullRow.collapsedData;
-      return true;
-    }
+    } else return true;
+  }
+
+  poolsFilterFunction(row: ViewGroupedPositions) {
+    const pools = this.dropDownFilters[1];
+    if (pools.selectedIndex == 0) return true;
+
+    const pool = this.poolNames[pools.selectedIndex - 1];
+    return pool.poolId === row.poolId;
   }
 
-  poolsFilterFunction(row: ViewProtectedLiquidity) {
-    return true;
+  get poolNames() {
+    const filteredNamedPos = this.groupedPos
+      .filter(
+        (obj, index, self) =>
+          index === self.findIndex(x => x.poolId === obj.poolId)
+      )
+      .map((value, index) => ({
+        id: index + 1,
+        title: buildPoolName(value.poolId),
+        poolId: value.poolId
```

Comment:
Using indexes as IDs in lists can defeat the purpose of giving array items IDs in the first place, as mutating the array doesn't help with any diffing in the background. 

We know a ```positionId``` will be unique, can we use that as an ID instead?